cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -232,93 +259,176 @@ runQueryUTxO (AnyConsensusModeParams cModeParams)
   maybeFiltered NoFilter = Nothing
 
 
+runQueryPoolParams
+  :: AnyCardanoEra
+  -> AnyConsensusModeParams
+  -> NetworkId
+  -> Hash StakePoolKey
+  -> ExceptT ShelleyQueryCmdError IO ()
+
+runQueryPoolParams anyEra@(AnyCardanoEra era) (AnyConsensusModeParams cModeParams)
+                    network poolid = do
+    SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
+
+    let consensusMode = consensusModeOnly cModeParams
```

Comment:
So this will work, but this can be refactored. If you look at my earlier comments that point out where refactoring was undone, it will be clearer as to how this can be refactored.